### PR TITLE
Adding section on shared variable control.

### PR DESCRIPTION
**Pull request by gusthoff**

### CODE REVIEW

**Comment by gusthoff** [comment]
A test issue, which affects two of the source-code examples found in this review, is fixed by #587.

File: content/courses/advanced-ada/chapters/types.rst [hunk 1]
```diff
@@ -3201,19 +3201,448 @@ majority of the cases, you don't need it at all |mdash| except for special
...
+hasn't changed the value. This situation may arise when an object in memory is
+being shared by multiple threads. For example, a thread *B* may modify the
+value of that object between two read accesses of a thread *A*. Another typical
+example is the one of
+`memory-mapped I/O <https://en.wikipedia.org/wiki/Memory-mapped_I/O>`_, where
+the hardware might be constantly changing the value of an object in memory.
 
-    Complete section!
+Because the value of a volatile object may be constantly changing, a compiler
+cannot generate code that stores the value of that object into a register and
```
Comment (pat-rogers) [inline content/courses/advanced-ada/chapters/types.rst:3224]: "cannot generate code that stores the value of that object into a register and" =>
"cannot generate code to store the value of that object in a register and then"

(This is a tricky bit of grammar.)

File: content/courses/advanced-ada/chapters/types.rst [hunk 2]
```diff
@@ -3201,19 +3201,448 @@ majority of the cases, you don't need it at all |mdash| except for special
...
+       R   : Integer
+         with Atomic, Address => System'To_Address (16#FFFF00A0#);
+
+    end Shared_Var_Types;
+
+Note that the :ada:`Address` aspect allows for assigning a variable to a
+specific location in the memory. In this example, we're using this aspect to
+specify the address of the memory-mapped register.
+
+In addition to atomic objects, we can declare atomic types |mdash| similarly to
```
Comment (pat-rogers) [inline content/courses/advanced-ada/chapters/types.rst:3569]: "similarly" => "similar"

File: content/courses/advanced-ada/chapters/types.rst (hunk 1 above)
Comment (gusthoff) [reply content/courses/advanced-ada/chapters/types.rst]: Thanks a lot!

**Review by pat-rogers — APPROVED** [review]

**Pull request merged**